### PR TITLE
[AMQ-9239] Host joram-jms-unit tests ahead of converting to jakarta.jms

### PR DESCRIPTION
**Pull request by mattrpav**
Currently this module is unmaintained and we need to convert it for jakarta broker support.

Modules that use this module:
activemq-amqp
activemq-unit-tests

ref: https://github.com/fusesource/joram-jms-tests

### CODE REVIEW

**Comment by gemmellr** [comment]
Personally I dont see enough value in these tests to copy+convert them, a lot of it is just testing basic stuff the rest of the tests should be covering themselves. I would simply drop their usage (and did exactly that elsewhere). They dont cover any of the newer stuff, not having been updated in over a decade. I cant remember a single time theyve actually shown a failure. The actual official spec tests are easily available now unlike back then if needed.

**Comment by mattrpav** [comment]
@gemmellr I think removing it would be a good end goal. However, right now its deeply integrated into amqp tests. I'll make a JIRA for removing joram and refactoring the amqp tests post-jakarta work.

**Comment by gemmellr** [comment]
Knowing how the Joram tests work, 'deeply integrated' seems a bit of an exageration. Theres one package of classes that could just be outright deleted, and it looks like there are a couple of other tests misusing some related code to enable broker protocol tracing that arguably shouldnt be enabled anyway and can be got other ways. Seems trivial to remove.

**Comment by mattrpav** [comment]
@gemmellr Yep, I've got AMQ-9249 tee'd up for after getting the Jakarta broker work in order. 

If you've got spare cycles now, go for it! I'm happy to rebase from there, if you can get it merged in before the Jakarta work is done.

Thanks

**Comment by gemmellr** [comment]
Just giving some feedback on a simpler approach, its not something I will be working on.

**Comment by mattrpav** [comment]
@gemmellr no worries, appreciated!

File: activemq-tooling/activemq-joram-jms-tests/src/main/java/org/objectweb/jtests/jms/conform/connection/TopicConnectionTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2009 Red Hat, Inc.
```
Comment (cshannon) [inline activemq-tooling/activemq-joram-jms-tests/src/main/java/org/objectweb/jtests/jms/conform/connection/TopicConnectionTest.java:2]: I noticed there is a Red Hat reference in some files like here, I assume we need to get rid of this?

Comment (mattrpav) [reply activemq-tooling/activemq-joram-jms-tests/src/main/java/org/objectweb/jtests/jms/conform/connection/TopicConnectionTest.java:2]: Already corrected in the jakarta PR commit:

https://github.com/apache/activemq/pull/996/commits/17d7f024873765ba05ac735f53dc321117dc8881

**Pull request merged**